pkg/instancegroups/instancegroups.go

```diff
@@ -561,3 +561,21 @@ func (c *RollingUpdateCluster) deleteNode(ctx context.Context, node *corev1.Node
 
 	return nil
 }
+
+func (c *RollingUpdateCluster) RollSingleInstance(ctx context.Context, cloudMember *cloudinstances.CloudInstanceGroupMember, detach bool) error {
+	if detach {
+		if cloudMember.CloudInstanceGroup.InstanceGroup.IsMaster() {
+			return fmt.Errorf("cannot detatch master instances. Run with --detatch=false")
+
+		}
+		err := c.detachInstance(cloudMember)
+		if err != nil {
+			return fmt.Errorf("failed to detach instance: %v", err)
+		}
+		if err := c.maybeValidate(" after detaching instance", c.ValidateCount); err != nil {
+			return err
+		}
+	}
+
+	return c.drainTerminateAndWait(ctx, cloudMember, false, 0)
```

Comment:
If the instance is a bastion, this will incorrectly try to drain it.

If the instance is a master, you might want to `maybeValidate` afterwards.